cmake/ProcessOptions.cmake

```
@@ -25,10 +25,23 @@ function( NEST_PROCESS_WITH_OPTIMIZE )
     if ( with-optimize STREQUAL "ON" )
```

There is one unintuitive path along wrongly-cased "on", "On" vs. "ON", which can be caught with `TOUPPER`
```suggestion
    string(TOUPPER "${with-optimize}" WITHOPTIMIZE)
    if ( WITHOPTIMIZE STREQUAL "ON" )
```